avocado/utils/memory.py

```diff
@@ -41,17 +41,17 @@ def _check_memory_state(block):
     Check the given memory block is online or offline
 
     :param block: memory block id.
-    :type string: like 198
+    :type string: like memory198
     :return: 'True' if online or 'False' if offline
     :rtype: bool
     """
     def _is_online():
-        with open('/sys/devices/system/memory/memory%s/state' % block, 'r') as state_file:
+        with open('/sys/devices/system/memory/%s/state' % block, 'r') as state_file:
             if state_file.read() == 'online\n':
                 return True
             return False
 
-    return wait.wait_for(_is_online, timeout=120, step=1) or False
+    return wait.wait_for(_is_online, timeout=2, step=1) or False
```

Comment:
Well this is quite a big change. Are you absolutely sure this is enough for __any__ system under whatever load? I'd suggest at least 10s.

As for the style, please separate this change to a different commit. That improves the readability of the changelog as well as possibility to revert/bisect. (in general when you need to use `also` in commit message, it means the commit is wrong and should be split).

personal note: I like `git cola` or `git citool` for splitting patches, but it's just a personal taste, feel free to use whatever tool that suits you.

Reply:
well, on my system I see it wont take much time to change the state with load. but it is practical to have 10s wait time.  I am thinking of a way to exit as soon as the state changes to 'online'  instead of waiting 10s even after the state has changed with in 1 second.  so in a case where this function is called for 100 iteration.. 10s x 100 adds up to too much of wait time.  do you suggest some way to exit as early as possible ?

Reply:
You're already using the wait.wait_for, which will interrupt the wait as soon as the `_is_online` returns `True`. With 1s steps, maximum wait period after the status changes to `online` is <1s. The `10s` is just the time we will keep trying, not the total time this call will interrupt the test.